examples/distributed_training/unet_training_horovod.py

```diff
@@ -0,0 +1,188 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This example shows how to execute distributed training based on Horovod APIs.
+It can run on several nodes with multiple GPU devices on every node.
+Main steps to set up the distributed training:
+
+- Install Horovod referring to the guide: https://github.com/horovod/horovod/blob/master/docs/gpus.rst
+  If using MONAI docker, which already has NCCL and MPI, can quickly install Horovod with command:
+  `HOROVOD_NCCL_INCLUDE=/usr/include HOROVOD_NCCL_LIB=/usr/lib/x86_64-linux-gnu HOROVOD_GPU_OPERATIONS=NCCL \
+  pip install --no-cache-dir horovod`
+- Set SSH permissions for root login without password at all nodes except master, referring to:
+  http://www.linuxproblem.org/art_9.html
+- Run `hvd.init()` to initialize Horovod.
+- Pin each GPU to a single process to avoid resource contention, use `hvd.local_rank()` to get GPU index.
+  And use `hvd.rank()` to get the overall rank index.
+- Wrap Dataset with `DistributedSampler`, and disable the `shuffle` and `num_worker=0` in DataLoader.
+  Instead, shuffle data by `train_sampler.set_epoch(epoch)` before every epoch.
+- Wrap the optimizer in hvd.DistributedOptimizer. The distributed optimizer delegates gradient
+  computation to the original optimizer, averages gradients using allreduce or allgather,
+  and then applies those averaged gradients.
+- Broadcast the initial variable states from rank 0 to all other processes.
+
+Note:
+    Suggest setting exactly the same software environment for every node, especially `mpi`, `nccl`, etc.
+    A good practice is to use the same MONAI docker image for all nodes directly, if using docker, need
+    to set SSH permissions both at the node and in docker, referring to Horovod guide for more details:
+    https://github.com/horovod/horovod/blob/master/docs/docker.rst
+
+    Example script to execute this program, only need to run on the master node:
+    `horovodrun -np 16 -H server1:4,server2:4,server3:4,server4:4 python unet_training_horovod.py -d "./testdata"`
+
+Referring to: https://github.com/horovod/horovod/blob/master/examples/pytorch_mnist.py
+
+"""
+
+import os
+import sys
+from glob import glob
+import nibabel as nib
+import numpy as np
+import torch
+import argparse
+from torch.utils.data.distributed import DistributedSampler
+import horovod.torch as hvd
+
+import monai
+from monai.transforms import (
+    Compose,
+    LoadNiftid,
+    AsChannelFirstd,
+    ScaleIntensityd,
+    RandCropByPosNegLabeld,
+    RandRotate90d,
+    ToTensord,
+)
+from monai.data import create_test_image_3d, Dataset, DataLoader
+
+
+def train(args):
+    # initialize Horovod library
+    hvd.init()
+    # Horovod limits CPU threads to be used per worker
+    torch.set_num_threads(1)
```

Comment:
is this needed:
https://github.com/horovod/horovod/blob/34baedf1257bbdc5de15e54723d6d7bc21409a87/examples/pytorch_mnist.py#L126-L130

Reply:
I tested both methods(num_worker=0 or 1), no obvious difference.
Anyway, I already updated to align with Horovod example in the latest commit.
Thanks.